sos/report/plugins/mellanox_firmware.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2023 Nvidia Corporation, Alin Serdean <aserdean@nvidia.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, SoSPredicate, IndependentPlugin
+import os
+import time
+
+
+class MellanoxFirmware(Plugin, IndependentPlugin):
+
+    short_desc = 'Nvidia(Mellanox) firmware tools output'
+
+    plugin_name = "mellanox_firmware"
+    profiles = ('hardware', 'system')
+
```

Comment:
As written, this will enable the plugin by default on every system that runs sos. I don't think that's intended, especially given the predicates below.

Adding at least a `packages` tuple here, and/or a `files`, `commands`, or `kernel_mods` will restrict the plugin to only running on systems that match those enablement triggers.

Reply:
Sorry, I missed `commands` option...
I added it as part of the incremental.

Regarding packages, `mft`, `mstflint` or both could be installed on the system...
I'm unsure how I could describe this as part of the `packages` tuple.

Reply:
You could potentially run a check at the beginning of `setup()` which determines which package is installed, and if both are installed then simply choose one as a default, and build a set of commands based on that result.